### PR TITLE
Fleming design model clean ups

### PR DESCRIPTION
**Pull request by jeanphilippeD**
Clean ups and rename that became necessary over time.

Rendered:
https://raw.githack.com/maidsafe/routing/01e821131a9562663c615bcc52c3e37578471e40/routing_model/index.html

### CODE REVIEW

File: routing_model/index.html [hunk 1]
```diff
@@ -642,41 +642,44 @@ <h2>Deciding that a member of our section should be relocated away</h2>
...
+  When we receive ExpectCandidateAcceptResponse or ExpectCandidateRefuseResponse RPC, we vote for it in PARSEC, regardless of the order of operations, so it will be consensused.<br/>
+  The first ParsecExpectCandidateAcceptResponse consensused for a node will be the single valid relocation. We will sign the reolcation info through parsec and send to the node that is being relocated the RelocatedInfo they will need.<br/>
+  At this point, we will purge their information since this node isn't a member of our section anymore.<br/>
   <br/>
-  Also of note: while we are preventing nodes with OnlineState from relocating when we are already relocating one of our Adult. We may still be actively relocating more than one node away from our section (for instance if two sections with a relocating node each merge).<br/>
-  However, we will only handle one at a time per CheckRelocateTimeOut event.
+  Elders are not considered for relocation: Elder nodes in RelocatingAgeIncreaseState will eventually get demoted to adult (see StartMergeSplitAndChangeElders) at which point they may be relocated.<br/>
```
Comment (pierrechevalier83) [inline routing_model/index.html]: "demoted to adult" -> "demoted to adults"

File: routing_model/index.html [hunk 2]
```diff
@@ -642,41 +642,44 @@ <h2>Deciding that a member of our section should be relocated away</h2>
...
+  We prioritize relocating our Adults, then just relocated node that need another hop, then nodes comming back online.<br/>
```
Comment (pierrechevalier83) [inline routing_model/index.html]: "just relocated node" -> "just relocated nodes"

File: routing_model/index.html [hunk 3]
```diff
@@ -169,7 +169,7 @@ <h2>Becoming a full member of a section</h2>
     StartComputeResourceProofForElder["start_compute_resource_proof(source elder)<br/><br/>schedule(TimeoutRefused)"]
     StartComputeResourceProofForElder --> LoopEnd
 
-    Rpc -- "ExpectCandidate<br/>RefuseCandidate<br/>RelocateResponse<br/>..." --> VoteParsecRPC
+    Rpc -- "ExpectCandidate<br/>ExpectCandidateRefuseResponse<br/>ExpectCandidateAcceptResponse<br/>..." --> VoteParsecRPC
```
Comment (pierrechevalier83) [inline routing_model/index.html]: I understand how the words: ExpectCandidateAcceptResponse and ExpectCandidateRefuseResponse came to be, but I find them a bit hard to parse. I think CandidateAccepted or CandidateRefused would read better, despite conveying a bit less explicitely that they are a response to the ExpectCandidate RPC.

In general, I would aim for short names that express just the amount of context needed to make sense of them, same as in code...

File: routing_model/index.html [hunk 4]
```diff
@@ -183,18 +182,18 @@ <h2>Node as a member of a section</h2>
...
   </ul>
-  All of these flows are happening simultaneously, but they share a common event loop. At any time, either all flows or all but one flows  must be in a "wait" state.<br/>
+  All of these flows are happening simultaneously, but they share a common event loop. At any time, either all flows or all but one flows  must be in a "wait" state. (If an event is handled by mutliple active event loops, the one with the highest number handles it.)<br/>
   If our section decides to relocates us, we will have to stop functioning as a member of our section and go back to the previous flow where we will "Rebootstrap" so we can become a member of a different section.
   </p>
   </div>
   <div class="mermaid">
     graph TB
-    style StartTopMember fill:#f9f,stroke:#333,stroke-width:4px
+    style StartSectionMemberNode fill:#f9f,stroke:#333,stroke-width:4px
```
[the remark this answers is not in the document]
Comment (jeanphilippeD) [reply routing_model/index.html]: Could be, one thing I liked about it is that it differentiated the infinite loops that do nothing until something happen and the other sub routines. I think we still probably want to keep a differentiation.

File: routing_model/index.html [hunk 5]
```diff
@@ -288,13 +287,13 @@ <h2>Deciding when to accept an incoming relocation</h2>
       RPC((RPC))
       RPC --ExpectCandidate--> VoteParsecExpectCandidate
 
-      ParsecConsensus((Parsec<br/>consensus))
+      ParsecConsensus((Consensus))
```
[the remark this answers is not in the document]
Comment (jeanphilippeD) [reply routing_model/index.html]: I made them consistent, the circle contain 'Consensus', the arrows says Parsec 'consensus'

File: routing_model/index.html [hunk 6]
```diff
@@ -368,7 +367,7 @@ <h2>Relocated node connection</h2>
 
       LoopStart-->WaitFor
 
-      WaitFor -- Consensus--> ParsecConsensus
+      WaitFor -- Parsec<br/>consensus--> ParsecConsensus
```
[the remark this answers is not in the document]
Comment (jeanphilippeD) [reply routing_model/index.html]: Same change, see other comment.

File: routing_model/index.html [hunk 7]
```diff
@@ -441,16 +440,16 @@ <h2>Relocated node connection</h2>
 
   <h2>Resource proof from a destination's point of view</h2>
   <div class=description>
-  <p>In the previous diagram, we ensured an incoming candidate would only reach the "WaitingProofing" state once it was connected to our section and able to communicate with its elders. At this stage, the node would be a member of our peer_list.<br/>
-  This is maintained accross merge/split as for any adult node, so we are ready to resource proof any node in WaitingProofing state<br/>
+  <p>In the previous diagram, we ensured an incoming candidate would only reach the "WaitingProofingState" state once it was connected to our section and able to communicate with its elders. At this stage, the node would be a member of our peer_list.<br/>
```
[the remark this answers is not in the document]
Comment (jeanphilippeD) [reply routing_model/index.html]: Sounds like a good idea.

File: routing_model/index.html (hunk 3 above)
Comment (jeanphilippeD) [reply routing_model/index.html]: Our RPCs in the code do have that format Name/NameResponse, If we want a shorter name, maybe we should look at the ExpectCandidate one. Could be `Relocating`, maybe.

Comment (pierrechevalier83) [reply routing_model/index.html]: Uhm. What about this:
`Rpc::ExpectCandidate`,
`Rpc::ExpectCandidateResponse`,
But then, like `ExpectCandidate` contains a `CandidateInfo`, `ExpectCandidateResponse` could contain a `Result<RelocationInfo>`, where `RelocationInfo` would be a `struct` containing the target range and the destination section info.

Comment (pierrechevalier83) [reply routing_model/index.html]: If the `result.is_err()`, of course it would mean a refusal. If that's not clear enough, we could have our own `enum RelocationResponse { Accepted, Refused };`

Comment (jeanphilippeD) [reply routing_model/index.html]: That was my original proposal, but when we discussed it we did not seem to like the nesting, I kind of come around to it.

Comment (pierrechevalier83) [reply routing_model/index.html]: I think it's OK with a caveat: we don't have to show the nesting on the graphs ;)
